client/site-performance/components/PageSelector.tsx

```diff
@@ -0,0 +1,92 @@
+import page from '@automattic/calypso-router';
+import { SearchableDropdown } from '@automattic/components';
+import { useDebouncedInput } from '@wordpress/compose';
+import { useI18n } from '@wordpress/react-i18n';
+import { useMemo } from 'react';
+import QueryPosts from 'calypso/components/data/query-posts';
+import { useSelector } from 'calypso/state';
+import { getPostsForQueryIgnoringPage } from 'calypso/state/posts/selectors';
+import getCurrentQueryArguments from 'calypso/state/selectors/get-current-query-arguments';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+interface Page {
+	slug: string;
+	URL: string;
+	title: string;
+}
+
+const PER_PAGE = 10;
+
+export const PageSelector = () => {
+	const { __ } = useI18n();
+
+	const queryParams = useSelector( getCurrentQueryArguments );
+	const site = useSelector( getSelectedSite );
+
+	const [ , setPageQuery, pageQuery ] = useDebouncedInput();
+	const query = { type: 'page', number: PER_PAGE, search: pageQuery, page: 1 };
+	const pages = useSelector( ( state ): Page[] | null =>
+		getPostsForQueryIgnoringPage( state, site?.ID, query )
+	);
+
+	const options = useMemo( () => {
+		const mappedPages =
+			pages?.map( ( page ) => {
+				let url = page.URL.replace( site?.URL ?? '', '' );
+				url = url.length > 1 ? url.replace( /\/$/, '' ) : url;
+
+				return {
+					URL: url,
+					label: page.title,
+					value: url,
+				};
+			} ) ?? [];
+
+		if ( pageQuery.length > 0 ) {
+			return mappedPages;
+		}
+
+		return [ { URL: '/', label: __( 'Home' ), value: '/' }, ...mappedPages ];
+	}, [ pageQuery.length, pages, site?.URL, __ ] );
+
+	return (
+		<>
+			<QueryPosts postId={ null } siteId={ site?.ID } query={ query } />
+			<SearchableDropdown
```

Comment:
From the PT it seems that we want to use the existing `<SuggestionSearch />` component instead of reusing the `<SearchableDropdown />`. But let me know what you think.

Reply:
From this design p9Jlb4-e2J-p2 it's recommended we use https://href.li/?https://wordpress.github.io/gutenberg/?path=/docs/components-comboboxcontrol--docs from core